app/assets/javascripts/initializers/initializeUserProfilePage.js

```diff
@@ -1,11 +1,23 @@
 'use strict';
 
 function initializeUserProfilePage() {
-  if (document.getElementById('user-profile-dropdown')) {
-    document.getElementById('user-profile-dropdown').onclick = () => {
-      document
-        .getElementById('user-profile-dropdownmenu')
-        .classList.toggle('showing');
-    };
+  if(document.querySelector('.profile-dropdown')){
+    var profileDropdownDiv = document.querySelector('.profile-dropdown');
+    var currentUser = userData();
+    var profileUser = JSON.parse(profileDropdownDiv.getAttribute('data-info'));
+
+    if(currentUser.username == profileUser.username){
+      profileDropdownDiv.hidden = true;
+    }
+    else{
+      profileDropdownDiv.hidden = false;
+      if (document.getElementById('user-profile-dropdown')) {
+        document.getElementById('user-profile-dropdown').onclick = () => {
+          document
+            .getElementById('user-profile-dropdownmenu')
+            .classList.toggle('showing');
+        };
+      }
```

Comment:
could you refactor this, with the same technique mentioned in the beginning? save `document.getElementById('user-profile-dropdown')` in a variable and then check it.

also possibly switch `element.onclick` to `element.addEventListener('click', function)`

thanks!